public/main.js

```diff
@@ -0,0 +1,68 @@
+// get the map - set location and zoom
+let mymap = L.map('map').setView([44.478, -73.211882], 16)
+L.tileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
+	maxZoom: 19,
+	attribution: '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
+}).addTo(mymap)
+
+let path = window.location.pathname
+let pathArray = path.split('/')
+let id = pathArray.pop()
+
+let listContainer = document.getElementById('idList')
+
+async function getData(id) {
+	let restList = await fetch('https://json-server.burlingtoncodeacademy.now.sh/restaurants')
+		.then((response) => {
+			return response.json()
+		})
+		.then((dataJson) => {
+			return dataJson
+		})
+
+	// populate sidebar
+	restList.forEach((post) => {
+		let id = post.id
+		let name = post.name
+		let address = post.address
+		let phone = post.phone
+		let webSite = post.webSite
+		let hours = post.hours
+		let notes = post.notes
+
+		listContainer.innerHTML += `<li><a href='/post/${id}'>${name}</a></li>`
+		getLatLon(address, id)
+
+		// put pins on map
+		async function getLatLon(address, restId) {
```

Comment:
since this function doesn't depend on any local variables in your getData() function, you can pull it out to the top level of your file. Other than that, this function is good!